### ISSUE TITLE
Configurable Source ID

### ISSUE THREAD

**Issue by ptrwtts** [reservoirprotocol/sample-marketplace#170]
Each marketplace can have a source ID, which is used to identify orders created on it. This is then used for filtering or showing an icon when browsing. We want to make sure that as people fork the sample marketplace, orders are being properly attributed.

The source ID must be a wallet address, which gets passed when posting new orders to the backend. The backend the maps the source ID to more information like Name, Image, URL. Long term, it would be good if developers could upload this info, but it's tricky, because then to allow editing, there needs to be authentication, etc. So for now, we add it manually. 

To allow marketplaces to show correct attribution info, even before added to the backend, here is the proposal:

1) Create an ENV variable for setting the Source ID
2) Pass it whenever publishing orders 
3) When the backend returns orders from this marketplace, they will have the ID, but be missing name, icon, etc. Do a local replacement with the real icon and name from ENV variables

**Comment by rorygeoghegan** [comment]
Slight change of plan from above.

1. If marketplace name has been set as an environment variable pass this as the source whenever publishing orders. Otherwise use Reservoir Market.

**Comment by diegoramosxyz** [comment]
@ptrwtts @ptrwtts Regarding task 3, which marketplaces are going to have locally?

**Comment by rorygeoghegan** [comment]
I don't believe this is required anymore
The backend will return the case sensitive source string (eg. OpenSea) and you should be able to then just display that as the name  and use these redirect apis to get icon and link 
https://api.reservoir.tools/#/5.%20Redirects

**Comment by ptrwtts** [comment]
@diegoramosxyz @rorygeoghegan 

So just to clarify:

- We now just use the name as the source ID, no more wallet addresses
- You can use the redirect API to get the logo + link for _external_ marketplaces
- For the _current_ marketplace, it should still use a local icon and link. Because in most cases, the backend won't have the icon and link yet. So we need a solution that works as soon as they deploy.

So to summarize:
- If source=ENV_NAME, then use local logo and link
- If source!=ENV_NAME, then use the redirect API
